app/controllers/application_controller.rb

```diff
@@ -2117,6 +2117,8 @@ def vm_publish
   def get_global_session_data
     # Set the current userid in the User class for this thread for models to use
     User.current_user = current_user
+    if current_user.try(:current_group_id_changed?)
+    end
```

Comment:
I guess this part is outstanding.

This lets us know that an outside process changed the user's current_group.
Not sure if we should check that they still have a valid group (and reset it) or what.